### PR TITLE
[gala_kinematic] fixing the loss computation

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

Addressing the issues raised here : https://github.com/facebookresearch/habitat-lab/pull/839/files#r840202226

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat/core/batched_env.py [hunk 1]
```diff
@@ -289,8 +289,8 @@ def get_nonpixel_observations(self, env_states, observations):
         # TODO: update observations here
         for (b, state) in enumerate(env_states):
             if self.include_point_goal_gps_compass:
-                robot_pos = state.robot_position
-                robot_yaw = state.robot_yaw
+                robot_pos = state.robot_pos
+                robot_yaw = 0.0 # state.robot_yaw
```
Comment (vincentpierre) [inline habitat/core/batched_env.py:293]: Not relevant to this change

File: habitat_baselines/common/rollout_storage.py [hunk 2]
```diff
@@ -77,9 +77,12 @@ def __init__(
             self.buffers["actions"] = self.buffers["actions"].long()
             self.buffers["prev_actions"] = self.buffers["prev_actions"].long()
 
-        self.buffers["masks"] = torch.zeros(
+        self.buffers["not_done_mask_0"] = torch.ones(
             numsteps + 1, num_envs, 1, dtype=torch.bool
         )
+        self.buffers["not_done_mask_1"] = torch.ones(
+            numsteps + 2, num_envs, 1, dtype=torch.bool
+        )
```
Comment (vincentpierre) [inline habitat_baselines/common/rollout_storage.py:85]: setting to True since the default is unmasked.

File: habitat_baselines/common/rollout_storage.py [hunk 3]
```diff
@@ -172,11 +191,12 @@ def compute_returns(self, next_value, use_gae, gamma, tau):
                     self.buffers["rewards"][step]
                     + gamma
                     * self.buffers["value_preds"][step + 1]
-                    * self.buffers["masks"][step + 1]
+                    * self.buffers["not_done_mask_0"][step + 1]
+                    * self.buffers["not_done_mask_1"][step + 1]
```
Comment (vincentpierre) [inline habitat_baselines/common/rollout_storage.py:195]: This is the same as logical_and. Only use next step return if both not_done_mask_0 and not_done_mask_1 are true (no reset)

File: habitat_baselines/rl/ppo/ppo.py [hunk 4]
```diff
@@ -146,9 +146,9 @@ def update(self, rollouts: RolloutStorage) -> Tuple[float, float, float]:
                     # Mask the loss. The mask corresponds to the transition right
                     # after a reset. This first observation needs to be ignored
                     # because its observation is actually part of the previous episode
-                    action_loss = action_loss * batch["masks"]
-                    value_loss = value_loss * batch["masks"]
-                    dist_entropy = dist_entropy * batch["masks"]
+                    action_loss = action_loss * batch["not_done_mask_0"]
```
Comment (vincentpierre) [inline habitat_baselines/rl/ppo/ppo.py:149]: Only mask if the first mask is false as per https://github.com/facebookresearch/habitat-lab/pull/839/files#r840202226

File: habitat_baselines/common/rollout_storage.py [hunk 5]
```diff
@@ -161,6 +177,9 @@ def after_update(self):
             0 for _ in self.current_rollout_step_idxs
         ]
 
+        self.buffers["not_done_mask_0"] = torch.ones_like(self.buffers["not_done_mask_0"])
+        self.buffers["not_done_mask_1"] = torch.ones_like(self.buffers["not_done_mask_1"])
+
```
Comment (erikwijmans) [inline habitat_baselines/common/rollout_storage.py:182]:
```suggestion
```

Don't reset the masks. You could reset on the slice `[2:]` but you need to carry information over as otherwise `[0]` will never get set.

Comment (erikwijmans) [reply habitat_baselines/common/rollout_storage.py:182]: I'll also put this here as I can't make a suggestion. But after
```
self.buffers[0] = self.buffers[self.current_rollout_step_idx]
```
You also need to do
```
self.buffers['not_done_mask_1'][1] = self.buffers['not_done_mask_1'][self.current_rollout_step_idx + 1]
```

**Review by erikwijmans — CHANGES_REQUESTED** [review]
Couple changes, but otherwise looks good.

**Review by erikwijmans — APPROVED** [review]

**Review by Skylion007 — APPROVED** [review]
